// ngx-timeline/dist/esm2015/ngx-timeline.js
import { Directive, TemplateRef, Component, Input, ContentChild, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * 自定义模板
 */
class TimelineDate {
    /**
     * @param {?} templateRef
     */
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
TimelineDate.decorators = [
    { type: Directive, args: [{ selector: 'ng-template[timelineDate]' },] },
];
/** @nocollapse */
TimelineDate.ctorParameters = () => [
    { type: TemplateRef, },
];
class TimelineHeader {
    /**
     * @param {?} templateRef
     */
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
TimelineHeader.decorators = [
    { type: Directive, args: [{ selector: 'ng-template[timelineHeader]' },] },
];
/** @nocollapse */
TimelineHeader.ctorParameters = () => [
    { type: TemplateRef, },
];
class TimelineBody {
    /**
     * @param {?} templateRef
     */
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
TimelineBody.decorators = [
    { type: Directive, args: [{ selector: 'ng-template[timelineBody]' },] },
];
/** @nocollapse */
TimelineBody.ctorParameters = () => [
    { type: TemplateRef, },
];
class TimelineFooter {
    /**
     * @param {?} templateRef
     */
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
TimelineFooter.decorators = [
    { type: Directive, args: [{ selector: 'ng-template[timelineFooter]' },] },
];
/** @nocollapse */
TimelineFooter.ctorParameters = () => [
    { type: TemplateRef, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxTimelineComponent {
    constructor() {
        this.dateFormat = 'yyyy-MM-dd';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NgxTimelineComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-timeline',
                template: `<div class="timeline" [ngClass]="timelineClass" [ngStyle]="timelineStyle">
  <div class="timeline-item" *ngFor="let event of events; index as i">
    <div class="timeline-date">
      <div *ngIf="!dateTemplate">{{event.date | date: dateFormat}}</div>
      <ng-template *ngIf="dateTemplate" [ngTemplateOutlet]="dateTemplate.templateRef" [ngTemplateOutletContext]="{ $implicit: event, rowIndex: i }"></ng-template>
    </div>
    <div class="timeline-point" [ngClass]="{'hide-border': event.icon}">
      <i class="fa" [ngClass]="event.icon"></i>
    </div>
    <div class="timeline-event">
      <div class="timeline-header">
        <div *ngIf="!headerTemplate">{{event.header}}</div>
        <ng-template *ngIf="headerTemplate" [ngTemplateOutlet]="headerTemplate.templateRef" [ngTemplateOutletContext]="{ $implicit: event, rowIndex: i }"></ng-template>
      </div>
      <div class="timeline-body">
        <div *ngIf="!bodyTemplate">{{event.body}}</div>
        <ng-template *ngIf="bodyTemplate" [ngTemplateOutlet]="bodyTemplate.templateRef" [ngTemplateOutletContext]="{ $implicit: event, rowIndex: i }"></ng-template>
      </div>
      <div class="timeline-footer">
        <div *ngIf="!footerTemplate">{{event.footer}}</div>
        <ng-template *ngIf="footerTemplate" [ngTemplateOutlet]="footerTemplate.templateRef" [ngTemplateOutletContext]="{ $implicit: event, rowIndex: i }"></ng-template>
      </div>
    </div>
  </div>
</div>`,
                styles: [`.timeline{position:relative}.timeline::before{position:absolute;width:2px;height:100%;top:0;left:50%;margin-left:-1px;background-color:#ccc;content:" "}.timeline .timeline-item{position:relative;margin:1rem 0}.timeline .timeline-item .timeline-date{-webkit-box-sizing:border-box;box-sizing:border-box;position:absolute;top:10px;left:55%;line-height:30px;width:45%}.timeline .timeline-item .timeline-point{-webkit-box-sizing:border-box;box-sizing:border-box;position:absolute;left:50%;top:10px;margin-left:-15px;border:1px solid #ccc;border-radius:100%;text-align:center;background-color:#fff;z-index:99;font-size:30px;height:30px;line-height:30px;width:30px}.timeline .timeline-item .timeline-point.hide-border{border:none}.timeline .timeline-item .timeline-event{-webkit-box-sizing:border-box;box-sizing:border-box;position:relative;left:0;width:45%;padding:1rem;border:1px solid #ccc;background-color:#fff;border-radius:5px}.timeline .timeline-item .timeline-event::after,.timeline .timeline-item .timeline-event::before{position:absolute;left:100%;top:10px;content:" ";height:0;width:0;border:15px solid transparent;border-left-color:#fff}.timeline .timeline-item .timeline-event::before{border-left-color:#ccc;margin-left:1px}.timeline .timeline-item .timeline-event .timeline-header{font-size:1.5rem;font-weight:500}.timeline .timeline-item .timeline-event .timeline-body{margin-top:1rem;margin-bottom:1rem;color:#666}.timeline .timeline-item:nth-child(even) .timeline-date{left:0;text-align:right}.timeline .timeline-item:nth-child(even) .timeline-event{left:55%}.timeline .timeline-item:nth-child(even) .timeline-event::after,.timeline .timeline-item:nth-child(even) .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item:nth-child(even) .timeline-event::before{border-right-color:#ccc;margin-left:-31px}@media (max-width:768px){.timeline::before{left:2%}.timeline .timeline-item{padding-top:30px}.timeline .timeline-item .timeline-date{top:0;left:7%;width:100%}.timeline .timeline-item .timeline-point{left:2%;top:40px}.timeline .timeline-item .timeline-event{left:7%;width:93%}.timeline .timeline-item .timeline-event::after,.timeline .timeline-item .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item .timeline-event::before{border-right-color:#ccc;margin-left:-31px}.timeline .timeline-item:nth-child(even) .timeline-date{text-align:left;left:7%}.timeline .timeline-item:nth-child(even) .timeline-event{left:7%}.timeline .timeline-item:nth-child(even) .timeline-event::after,.timeline .timeline-item:nth-child(even) .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item:nth-child(even) .timeline-event::before{border-right-color:#ccc;margin-left:-31px}}`]
            },] },
];
/** @nocollapse */
NgxTimelineComponent.ctorParameters = () => [];
NgxTimelineComponent.propDecorators = {
    "events": [{ type: Input },],
    "timelineClass": [{ type: Input },],
    "timelineStyle": [{ type: Input },],
    "dateFormat": [{ type: Input },],
    "dateTemplate": [{ type: ContentChild, args: [TimelineDate,] },],
    "headerTemplate": [{ type: ContentChild, args: [TimelineHeader,] },],
    "bodyTemplate": [{ type: ContentChild, args: [TimelineBody,] },],
    "footerTemplate": [{ type: ContentChild, args: [TimelineFooter,] },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxTimelineModule {
}
NgxTimelineModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    NgxTimelineComponent,
                    TimelineDate,
                    TimelineHeader,
                    TimelineBody,
                    TimelineFooter
                ],
                exports: [
                    NgxTimelineComponent,
                    TimelineDate,
                    TimelineHeader,
                    TimelineBody,
                    TimelineFooter
                ]
            },] },
];
/** @nocollapse */
NgxTimelineModule.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { NgxTimelineModule, NgxTimelineComponent as ɵa, TimelineBody as ɵd, TimelineDate as ɵb, TimelineFooter as ɵe, TimelineHeader as ɵc };
