// ngx-timeline/dist/bundles/ngx-timeline.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common'], factory) :
	(factory((global['ngx-timeline'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,core,common) { 'use strict';

var TimelineDate = /** @class */ (function () {
    function TimelineDate(templateRef) {
        this.templateRef = templateRef;
    }
    return TimelineDate;
}());
TimelineDate.decorators = [
    { type: core.Directive, args: [{ selector: 'ng-template[timelineDate]' },] },
];
TimelineDate.ctorParameters = function () { return [
    { type: core.TemplateRef, },
]; };
var TimelineHeader = /** @class */ (function () {
    function TimelineHeader(templateRef) {
        this.templateRef = templateRef;
    }
    return TimelineHeader;
}());
TimelineHeader.decorators = [
    { type: core.Directive, args: [{ selector: 'ng-template[timelineHeader]' },] },
];
TimelineHeader.ctorParameters = function () { return [
    { type: core.TemplateRef, },
]; };
var TimelineBody = /** @class */ (function () {
    function TimelineBody(templateRef) {
        this.templateRef = templateRef;
    }
    return TimelineBody;
}());
TimelineBody.decorators = [
    { type: core.Directive, args: [{ selector: 'ng-template[timelineBody]' },] },
];
TimelineBody.ctorParameters = function () { return [
    { type: core.TemplateRef, },
]; };
var TimelineFooter = /** @class */ (function () {
    function TimelineFooter(templateRef) {
        this.templateRef = templateRef;
    }
    return TimelineFooter;
}());
TimelineFooter.decorators = [
    { type: core.Directive, args: [{ selector: 'ng-template[timelineFooter]' },] },
];
TimelineFooter.ctorParameters = function () { return [
    { type: core.TemplateRef, },
]; };
var NgxTimelineComponent = /** @class */ (function () {
    function NgxTimelineComponent() {
        this.dateFormat = 'yyyy-MM-dd';
    }
    NgxTimelineComponent.prototype.ngOnInit = function () {
    };
    return NgxTimelineComponent;
}());
NgxTimelineComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'ngx-timeline',
                template: "<div class=\"timeline\" [ngClass]=\"timelineClass\" [ngStyle]=\"timelineStyle\">\n  <div class=\"timeline-item\" *ngFor=\"let event of events; index as i\">\n    <div class=\"timeline-date\">\n      <div *ngIf=\"!dateTemplate\">{{event.date | date: dateFormat}}</div>\n      <ng-template *ngIf=\"dateTemplate\" [ngTemplateOutlet]=\"dateTemplate.templateRef\" [ngTemplateOutletContext]=\"{ $implicit: event, rowIndex: i }\"></ng-template>\n    </div>\n    <div class=\"timeline-point\" [ngClass]=\"{'hide-border': event.icon}\">\n      <i class=\"fa\" [ngClass]=\"event.icon\"></i>\n    </div>\n    <div class=\"timeline-event\">\n      <div class=\"timeline-header\">\n        <div *ngIf=\"!headerTemplate\">{{event.header}}</div>\n        <ng-template *ngIf=\"headerTemplate\" [ngTemplateOutlet]=\"headerTemplate.templateRef\" [ngTemplateOutletContext]=\"{ $implicit: event, rowIndex: i }\"></ng-template>\n      </div>\n      <div class=\"timeline-body\">\n        <div *ngIf=\"!bodyTemplate\">{{event.body}}</div>\n        <ng-template *ngIf=\"bodyTemplate\" [ngTemplateOutlet]=\"bodyTemplate.templateRef\" [ngTemplateOutletContext]=\"{ $implicit: event, rowIndex: i }\"></ng-template>\n      </div>\n      <div class=\"timeline-footer\">\n        <div *ngIf=\"!footerTemplate\">{{event.footer}}</div>\n        <ng-template *ngIf=\"footerTemplate\" [ngTemplateOutlet]=\"footerTemplate.templateRef\" [ngTemplateOutletContext]=\"{ $implicit: event, rowIndex: i }\"></ng-template>\n      </div>\n    </div>\n  </div>\n</div>",
                styles: [".timeline{position:relative}.timeline::before{position:absolute;width:2px;height:100%;top:0;left:50%;margin-left:-1px;background-color:#ccc;content:\" \"}.timeline .timeline-item{position:relative;margin:1rem 0}.timeline .timeline-item .timeline-date{-webkit-box-sizing:border-box;box-sizing:border-box;position:absolute;top:10px;left:55%;line-height:30px;width:45%}.timeline .timeline-item .timeline-point{-webkit-box-sizing:border-box;box-sizing:border-box;position:absolute;left:50%;top:10px;margin-left:-15px;border:1px solid #ccc;border-radius:100%;text-align:center;background-color:#fff;z-index:99;font-size:30px;height:30px;line-height:30px;width:30px}.timeline .timeline-item .timeline-point.hide-border{border:none}.timeline .timeline-item .timeline-event{-webkit-box-sizing:border-box;box-sizing:border-box;position:relative;left:0;width:45%;padding:1rem;border:1px solid #ccc;background-color:#fff;border-radius:5px}.timeline .timeline-item .timeline-event::after,.timeline .timeline-item .timeline-event::before{position:absolute;left:100%;top:10px;content:\" \";height:0;width:0;border:15px solid transparent;border-left-color:#fff}.timeline .timeline-item .timeline-event::before{border-left-color:#ccc;margin-left:1px}.timeline .timeline-item .timeline-event .timeline-header{font-size:1.5rem;font-weight:500}.timeline .timeline-item .timeline-event .timeline-body{margin-top:1rem;margin-bottom:1rem;color:#666}.timeline .timeline-item:nth-child(even) .timeline-date{left:0;text-align:right}.timeline .timeline-item:nth-child(even) .timeline-event{left:55%}.timeline .timeline-item:nth-child(even) .timeline-event::after,.timeline .timeline-item:nth-child(even) .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item:nth-child(even) .timeline-event::before{border-right-color:#ccc;margin-left:-31px}@media (max-width:768px){.timeline::before{left:2%}.timeline .timeline-item{padding-top:30px}.timeline .timeline-item .timeline-date{top:0;left:7%;width:100%}.timeline .timeline-item .timeline-point{left:2%;top:40px}.timeline .timeline-item .timeline-event{left:7%;width:93%}.timeline .timeline-item .timeline-event::after,.timeline .timeline-item .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item .timeline-event::before{border-right-color:#ccc;margin-left:-31px}.timeline .timeline-item:nth-child(even) .timeline-date{text-align:left;left:7%}.timeline .timeline-item:nth-child(even) .timeline-event{left:7%}.timeline .timeline-item:nth-child(even) .timeline-event::after,.timeline .timeline-item:nth-child(even) .timeline-event::before{left:0;border-left-color:transparent!important;border-right-color:#fff;margin-left:-30px}.timeline .timeline-item:nth-child(even) .timeline-event::before{border-right-color:#ccc;margin-left:-31px}}"]
            },] },
];
NgxTimelineComponent.ctorParameters = function () { return []; };
NgxTimelineComponent.propDecorators = {
    "events": [{ type: core.Input },],
    "timelineClass": [{ type: core.Input },],
    "timelineStyle": [{ type: core.Input },],
    "dateFormat": [{ type: core.Input },],
    "dateTemplate": [{ type: core.ContentChild, args: [TimelineDate,] },],
    "headerTemplate": [{ type: core.ContentChild, args: [TimelineHeader,] },],
    "bodyTemplate": [{ type: core.ContentChild, args: [TimelineBody,] },],
    "footerTemplate": [{ type: core.ContentChild, args: [TimelineFooter,] },],
};
var NgxTimelineModule = /** @class */ (function () {
    function NgxTimelineModule() {
    }
    return NgxTimelineModule;
}());
NgxTimelineModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    NgxTimelineComponent,
                    TimelineDate,
                    TimelineHeader,
                    TimelineBody,
                    TimelineFooter
                ],
                exports: [
                    NgxTimelineComponent,
                    TimelineDate,
                    TimelineHeader,
                    TimelineBody,
                    TimelineFooter
                ]
            },] },
];
NgxTimelineModule.ctorParameters = function () { return []; };

exports.NgxTimelineModule = NgxTimelineModule;
exports.ɵa = NgxTimelineComponent;
exports.ɵd = TimelineBody;
exports.ɵb = TimelineDate;
exports.ɵe = TimelineFooter;
exports.ɵc = TimelineHeader;

Object.defineProperty(exports, '__esModule', { value: true });

})));
